### PR TITLE
V2

### PR DESCRIPTION
**Pull request by khenam**
Adjust to compatibility and concept change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,8 +131,10 @@ VueGoogleMap.load({
 Vue.component('google-map', VueGoogleMap.Map);
 new Vue({
     el: 'body',
-});
-
+    data:{
```
Comment (darrynten) [inline README.md:134]: Formatting is inconsistent

File: package.json [hunk 2]
```diff
@@ -1,17 +1,17 @@
 {
   "name": "vue-google-maps",
-  "version": "0.1.21",
+  "version": "1.0.0",
```
Comment (darrynten) [inline package.json:3]: This is cool, but the doesn't the actual name have to change, as it's no longer the same as the repo it was forked from?

File: src/components/rectangle.vue [hunk 3]
```diff
@@ -41,42 +57,94 @@ const events = [
     'mouseup',
     'rightclick'
 ]
-
+const getLocalField = function (self, field){
+  return (typeof self.$options.propsData[field] !== 'undefined')?self[field]:self.rectangleObj[field];
```
Comment (darrynten) [inline src/components/rectangle.vue:61]: Your formatting is inconsistent with the rest of the project

File: src/components/rectangle.vue [hunk 4]
```diff
@@ -1,20 +1,37 @@
...
+</template>
+
 <script>
 
 import _ from 'lodash';
 
-import eventBinder from '../utils/eventsBinder.js'
-import propsBinder from '../utils/propsBinder.js'
+import eventBinder from '../utils/eventsBinder.js';
+import propsBinder from '../utils/propsBinder.js';
```
Comment (darrynten) [inline src/components/rectangle.vue:11]: You don't need semi-colons with es6

File: src/components/infoWindow.vue [hunk 5]
```diff
@@ -44,37 +40,129 @@ const props = {
...
+      }
+    },
+    local_opened:{
+      get(){
+        return getLocalField(this, 'opened');
+      },
+      set(value){
+        this.infoWindowObj.opened = value;
+        this.$emit('opened_changed', value);
+        this.$nextTick(function (){
```
Comment (darrynten) [inline src/components/infoWindow.vue:115]: Shouldn't you be using a lexical this here?

```js
this.$nextTick( () => {
```

to preserve context?

I may be completely wrong though depending on how $nextTick works internally

File: src/components/infoWindow.vue [hunk 6]
```diff
@@ -44,37 +40,129 @@ const props = {
...
       }
       innerChanged();
       this.disconnect = mutationObserver(this.$el, innerChanged);
     } 
   },
 
   deferredReady() {
-    this.$dispatch('register-infoWindow', this);
+    var parent = this.getParentAcceptInfoWindow(this);
+    if (parent)
```
Comment (darrynten) [inline src/components/infoWindow.vue:164]: Single line ifs should be wrapped in `{}` like the rest of the project

**Review by darrynten — COMMENTED** [review]
A few notes here and there, but it's looking good.

I will be trying this PR out later today and will report back if I find anything fundamentally wrong

**Comment by darrynten** [comment]
Conflicts need resolution

**Comment by khenam** [comment]
Sorry about that, i will open a new PR and i will work on your suggestions, i really made a mistake with format and syntax.

**Pull request closed without merging**